bittensor/_subtensor/subtensor_impl.py

```diff
@@ -285,56 +227,63 @@ def subscribe(
+                    return False
             else:
-                logger.error( "Failed to subscribe")
-            return result
+                return True
 
     def add_stake(
             self, 
             wallet: 'bittensor.wallet',
             amount: Balance, 
-            hotkey_id: int, 
             wait_for_inclusion: bool = False,
             wait_for_finalization: bool = False,
-            timeout: int = 3 * bittensor.__blocktime__,
         ) -> bool:
         r""" Adds the specified amount of stake to passed hotkey uid.
         Args:
             wallet (bittensor.wallet):
                 bittensor wallet object.
             amount (bittensor.utils.balance.Balance):
                 amount to stake as bittensor balance
-            hotkey_id (int):
-                uid of hotkey to stake into.
             wait_for_inclusion (bool):
                 if set, waits for the extrinsic to enter a block before returning true, 
                 or returns false if the extrinsic fails to enter the block within the timeout.   
             wait_for_finalization (bool):
                 if set, waits for the extrinsic to be finalized on the chain before returning true,
                 or returns false if the extrinsic fails to be finalized within the timeout.
-            timeout (int):
-                time that this call waits for either finalization of inclusion.
         Returns:
             success (bool):
                 flag is true if extrinsic was finalized or uncluded in the block. 
                 If we did not wait for finalization / inclusion, the response is true.
         """
-        
         with self.substrate as substrate:
             call = substrate.compose_call(
-                call_module='SubtensorModule',
+                call_module='SubtensorModule', 
                 call_function='add_stake',
                 call_params={
-                    'hotkey': hotkey_id,
+                    'hotkey': wallet.hotkey.public_key,
                     'ammount_staked': amount.rao
                 }
             )
             extrinsic = substrate.create_signed_extrinsic( call = call, keypair = wallet.coldkey )
-            return self._submit_and_check_extrinsic ( extrinsic, wait_for_inclusion, wait_for_finalization, timeout )
+            response = substrate.submit_extrinsic( extrinsic, wait_for_inclusion = wait_for_inclusion, wait_for_finalization = wait_for_finalization )
+            if wait_for_inclusion or wait_for_finalization:
+                response.process_events()
+                if response.is_success:
+                    bittensor.logging.success( 'Staked', str(amount.tao) )
+                    return True
+                else:
+                    bittensor.logging.warning( 'Failed to stake', str(response.error_message) )
+                    return False
+            else:
+                return True
```

Comment:
Don't think we need an else here, can just return True. Same for the else on line 282.

Reply:
mega nit